Add the datasource of RDBMS (PostgreSQL and MySQL)

anzai opened the pull request:
I added the datasource of RDBMS named "SQL DB". This plugin supports PostgreSQL and MySQL.
- This datasource is implemented refering to the one of influxDB. So the all features with influxDB are also supported in SQL DB datasource. You can use query editor, defining raw query, templating, annotation, mixed queries, search condition with regex, and so on.
- The panel interface of frontend is very similar to influxDB datasource.
- SQL DB datasource needs the backend to connect databases of RDBMS, because we cannot connect RDBMS via HTTP.
  The backend listens HTTP POSTs from the frontend, connects databases and execute queries, and returns the results to the frontend.
  In this plugin, the backend uses xorm to connect RDBMS.
- SQL DB datasource needs users to specify which column has timestamp data.

Here is the detail about the fearures of SQL DB datasource.
https://github.com/anzai/grafana/wiki/SQL-DB-plugin-%28datasource%29

Commit:
Add the datasource of RDBMS (PostgreSQL and MySQL)

Commit:
Remove the extra variable

tokuhirom commented on pkg/api/sqldb/sqldb.go:
```diff
@@ -0,0 +1,165 @@
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/grafana/grafana/pkg/middleware"
+	m "github.com/grafana/grafana/pkg/models"
+	// "github.com/grafana/grafana/pkg/util"
```
trash?

anzai replied:
I removed it.

Commit:
Remove the extra line

torkelo commented:
this looks great! hope to have time to review it soon. 

The frontend data source plugin will probably need to go into it's own repo and be an external plugin that you install. But the backend part can be part of grafana core

anzai commented:
@torkelo Thank you. I think so, too. The JS part might better to be an external plugin,  and I'll be glad to maintain it on my own.

Commit:
Add support of "date" data type

volter commented:
anzai, thank you very much for your work! I tried it a bit with PostgreSQL and like it a lot. I found a few things that could be improved and I'll try to provide feedback somewhere. Not sure where though!

Commit:
Fix to get schema names from information schema in some version of PostgreSQL
In PostgreSQL, with the version before 9.4, unprivileged users
cannot get schema list from information_schama.schemata.

Commit:
Fix $timeFilter in MySQL

anzai commented:
@torkelo Could I separate the datasoure plugin from Grafana?
I'm just starting to try it, but grunt-typescript fails because the datasource plugin refers to some files in public/app of Grafana. Another published external plugins aren't written with TypeScript or don't import Grafana's files. I have no idea what I should do in such case...

torkelo commented:
@anzai great that you are trying to move the frontend part to a plugin. 

This is actually the first plugin that will use typescript and will hit this error. We have had similar issues with other plugins (how to get unit tests running that depend on grafana core files) but have yet found a solid solution. 

One solution that we have just started test and not yet fully worked out yet is to have a npm sdk package with the sass and core js/ts files. 

If you have time you can test get the external plugin typescript to compile by referring to 
https://www.npmjs.com/package/grafana-sdk-test 

The tricky thing (I have not tried this myself yet), is to make the typescript compiler happy with the paths and that the paths work at compile and at runtime

Commit:
Add parseFloat() to return table format

Commit:
Fix to show text values in tabel panel

anzai commented:
@torkelo Thank you for your hint. I succeeded to separate the frontend.
This is the temporary repository: https://github.com/anzai/grafana-sqldb
(If this will be published actually, I will create the repository as my company's one.)

Would I send another pull request including only the backend part?

torkelo commented:
@anzai yes, that might be best as then you could continue to improve and polish the frontend part without a PR changing

bergquist commented:
Closing this since #5364 was created.

liuzongquan commented:
@anzai could you provide an introduction for installing grafana-sqldb?

anzai commented:
@liuzongquan Sorry, but could you read in the newer PR https://github.com/grafana/grafana/pull/5364 ?

You can try the features with this binary including the patched backend and frontend plugin.
https://github.com/sraoss/grafana-sqldb-datasource/releases/download/V0.1/grafana-sqldb_20160615.tar.gz

liuzongquan commented:
@anzai Thanks a lot, anzai~
